src/linalg/impl_linalg.rs

```diff
@@ -588,8 +588,8 @@ pub fn general_mat_vec_mul<A, S1, S2, S3>(alpha: A,
                     {
                         let a_trans = CblasNoTrans;
                         let a_stride = match layout {
```

Comment:
It would be nice to add a comment, e.g.

```rust
                        // Determine stride between rows or columns. Note that the stride is
                        // adjusted to at least `k` or `m` to handle the case of a matrix with a
                        // trivial (length 1) dimension, since the stride for the trivial dimension
                        // may be arbitrary.
```